microcosm_postgres/factories/sessionmaker.py

```diff
@@ -1,9 +1,47 @@
+from microcosm.api import defaults
+from microcosm.config.types import boolean
+from microcosm.config.validation import typed
+from microcosm.scoping.factories import ScopedFactory
+from microcosm.scoping.proxies import ScopedProxy
+from sqlalchemy.orm import Session
 from sqlalchemy.orm import sessionmaker
 
 
+ENGINE = "__engine__"
+
+
+def unwrap(bind):
```

Comment:
Note that we have to `unwrap` proxies.

Note further that even though the `ScopedFactory` usage on L27 is conditional, the way `ScopedFact.infect` is implemented is... well... infectious and doesn't "undo" when we reset unit test context. As a result, we need the `unwrap` ability even in unrelated tests. This was annoying to debug.